atlasdb-commons/src/main/java/com/palantir/common/concurrent/InterruptibleFuture.java

```diff
@@ -113,10 +113,14 @@ public final boolean cancel(boolean mayInterruptIfRunning) {
         return futureTask.cancel(mayInterruptIfRunning);
     }
 
-    @SuppressWarnings("GuardedByChecker")
     protected void noteFinished() {
-        state = State.COMPLETED;
-        condition.signalAll();
+        lock.lock();
+        try {
+            state = State.COMPLETED;
+            condition.signalAll();
+        } finally {
+            lock.unlock();
+        }
     }
```

Comment:
Alternatively I think we can annotate this method with `@GuardedBy("lock")` to ensure the lock is already held.

Reply:
good call and simplifies things